source/session.ts

```diff
@@ -962,6 +970,10 @@ export class Session {
     let url: string;
     let headers: Record<string, string> = {};
 
+    options.signal?.addEventListener("abort", () => {
+      xhr.abort();
+    });
```

Comment:
should we remove the event listener as soon as the abort is triggered?

Reply:
[144c338] Remove abort handler once fired once